case_studies/dc2/DC2_bootstrap_generate_data.py

```diff
@@ -0,0 +1,177 @@
+        assert (
+            split_result["images"].shape[1] == split_result["images"].shape[2]
+        ), "image width should be equal to image height"
+
+        cur_plocs, cur_source_type, cur_flux = get_lsst_params(
+            lsst_catalog_tensors_dict,
+            cur_image_wcs,
+            image_lim,
+            image_height_index=split_result["image_height_index"],
+            image_width_index=split_result["image_width_index"],
+        )
+
+        with open(output_path / ("lsst_" + split_id), "wb") as output_file:
+            torch.save(
+                {
+                    "plocs": cur_plocs.clone(),
+                    "source_type": cur_source_type.clone(),
+                    "flux": cur_flux.clone(),
+                },
+                output_file,
+            )
+
+
+if __name__ == "__main__":
+    print("+" * 100, flush=True)
+    print("initialization begins", flush=True)
+    environ["BLISS_HOME"] = str(Path().resolve().parents[1])
+
+    output_dir = Path("./DC2_bootstrap_output/")
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    with initialize(config_path=".", version_base=None):
+        notebook_cfg = compose("notebook_config")
+    print("initialization ends", flush=True)
+    print("+" * 100, flush=True)
+
+    print("+" * 100, flush=True)
+    print("load dc2", flush=True)
+    dc2: DC2 = instantiate(notebook_cfg.surveys.dc2)
+    dc2.prepare_data()
+    dc2.setup()
+    dc2_test_dataset = dc2.test_dataset
+    print("+" * 100, flush=True)
+
+    print("+" * 100, flush=True)
+    print("load lsst catalog", flush=True)
+    GCRCatalogs.set_root_dir("/data/dc2/")
+    lsst_catalog_gcr = GCRCatalogs.load_catalog("desc_dc2_run2.2i_dr6_object_with_truth_match")
+    lsst_catalog_sub = lsst_catalog_gcr.get_quantities(
+        [
+            "id_truth",
+            "objectId",
+            "ra",
+            "dec",
+            "truth_type",
+            "cModelFlux_u",
+            "cModelFluxErr_u",
+            "cModelFlux_g",
+            "cModelFluxErr_g",
+            "cModelFlux_r",
+            "cModelFluxErr_r",
+            "cModelFlux_i",
+            "cModelFluxErr_i",
+            "cModelFlux_z",
+            "cModelFluxErr_z",
+            "cModelFlux_y",
+            "cModelFluxErr_y",
+        ]
+    )
+    lsst_catalog_df = pd.DataFrame(lsst_catalog_sub)
+    lsst_catalog_tensors_dict = {
+        "truth_type": torch.tensor(lsst_catalog_df["truth_type"].values).view(-1, 1),
+        "flux": torch.cat(
+            [
+                torch.tensor(flux.values).view(-1, 1)
+                for flux in [
+                    lsst_catalog_df["cModelFlux_g"],
+                    lsst_catalog_df["cModelFlux_i"],
+                    lsst_catalog_df["cModelFlux_r"],
+                    lsst_catalog_df["cModelFlux_u"],
+                    lsst_catalog_df["cModelFlux_y"],
+                    lsst_catalog_df["cModelFlux_z"],
+                ]
```

Comment:
Please use an intermediate variable here, to avoid deep nesting, which is hard to read

Reply:
fixed it